### PR TITLE
Babel and Webpack updates

### PR DESCRIPTION
**Pull request by chinmaykunkikar**
- Match the node version with the one we are using in docker environment (`v16`)
   Webpack doesn't recognize `"maintained node versions"` browserslist query for node target and throws errors.

 - Update the browerslist query string for more recent browsers.
   `~0.2%` audience uses Firefox versions between `v44` and `v100` - https://browsersl.ist/#q=%3E+0.2%25+and+not+dead%2C+firefox+%3E%3D+44
   So we can safely use newer versions of Firefox with the inclusion of Firefox ESR versions (`v78`). This browserslist query has a `~94%` coverage. This can be verified here -
   https://browsersl.ist/#q=%3E+0.2%25%2C+last+2+versions%2C+not+dead%2C+Firefox+ESR

 - Run `npx update-browserslist-db@latest` to install the latest `caniuse-lite` version (`1.0.30001431`).
 
   We see this suggestion in webpack logs all the time -
    >    static_builder_1    | Browserslist: caniuse-lite is outdated. Please run:
    >    static_builder_1    | npx browserslist@latest --update-db
    >    static_builder_1    |
    >    static_builder_1    | Why you should do it regularly:
    >    static_builder_1    | https://github.com/browserslist/browserslist#browsers-data-updating

   ...not any more.

 - We also see a tiny improvement in bundling size -

Before	| 	After
:---------:|:---------:
 | 
---
Other changes -
 - Update babel runtime and @babel/core to `v7.20`.
 - Update babel presets and plugins to latest versions.
 - Update webpack to `v5.75.0`.
 - Update webpack plugins to latest versions.

### CODE REVIEW

**Comment by amCap1712** [comment]
This would also fix SEC-902

File: babel.config.js [hunk 1]
```diff
@@ -12,10 +12,10 @@ module.exports = {
       "@babel/preset-env",
       {
         useBuiltIns: "usage",
-        corejs: { version: "3.9", proposals: true },
+        corejs: { version: "3.26", proposals: true },
         targets: {
-          node: "12",
-          browsers: ["> 0.2% and not dead", "firefox >= 44"],
+          node: "16",
+          browsers: ["> 0.2%", "last 2 versions", "not dead", "Firefox ESR"],
```
Comment (MonkeyDo) [inline babel.config.js:18]: Our contributor CatQuest uses FireFox 44, so I'd like to see that version kept in the list.

**Review by MonkeyDo — COMMENTED** [review]
Apart from the comment  below, looks good to me !

And sorry, I made a mess with the React update PR and now you have to solve a merge conflict with package.json and then re-run `npm install` to re-generated the package-lock file  😅

File: babel.config.js (hunk 1 above)
Comment (chinmaykunkikar) [reply babel.config.js:18]: Alright, I'm on it

**Comment by chinmaykunkikar** [comment]
@MonkeyDo all yours :D

**Pull request merged**